### PR TITLE
Relayer: add new endpoint

### PR DESCRIPTION
**Pull request by JustynaBroniszewska**

### CODE REVIEW

File: universal-login-contracts/lib/deployFactory.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import {Wallet, ContractFactory} from 'ethers';
+import {TransactionOverrides} from '@universal-login/commons';
+import {getDeployData} from './encode';
+import ProxyContract from '../build/Proxy.json';
+import ProxyCounterfactualFactory from '../build/ProxyCounterfactualFactory.json';
+
+export const deployFactory = async (wallet: Wallet, walletMasterAddress: string, overrideOptions?: TransactionOverrides) => {
+  const initCode = getDeployData(ProxyContract as any, [walletMasterAddress, '0x0']);
```
Comment (marekkirejczyk) [inline universal-login-contracts/lib/deployFactory.ts]: `ProxyContract as ContractJSON`

File: universal-login-contracts/test/contracts/ProxyCounterfactualFactory.ts [hunk 2]
```diff
@@ -27,8 +27,8 @@ describe('Counterfactual Factory', () => {
     ({deployer, ensDomainData, walletMaster, provider} = await loadFixture(ensAndMasterFixture));
     [wallet, anotherWallet] = getWallets(provider);
     [, initializeWithENS] = createProxyDeployWithENSArgs(keyPair.publicKey, ensDomainData, walletMaster.address);
+    factoryContract = await deployFactory(deployer, walletMaster.address);
```
Comment (marekkirejczyk) [inline universal-login-contracts/test/contracts/ProxyCounterfactualFactory.ts]: Shall we move factoryContract to fixture as well?

File: universal-login-relayer/lib/relayer.ts [hunk 3]
```diff
@@ -65,7 +65,7 @@ class Relayer {
     }));
     this.ensService = new ENSService(this.config.chainSpec.ensAddress, this.config.ensRegistrars, this.provider);
     this.authorisationService = new AuthorisationService(this.database);
-    this.walletContractService = new WalletService(this.wallet, this.config.walletMasterAddress, this.ensService, this.hooks);
+    this.walletContractService = new WalletService(this.wallet, this.config.walletMasterAddress, this.ensService, this.hooks, this.config.factoryAddress);
```
Comment (marekkirejczyk) [inline universal-login-relayer/lib/relayer.ts]: I would consider passing `this.config` instead of both `this.config.walletMasterAddress` and `this.config.factoryAddress`.

File: universal-login-relayer/lib/services/WalletService.ts [hunk 4]
```diff
@@ -33,6 +36,17 @@ class WalletService {
     }
     throw new InvalidENSDomain(ensName);
   }
+
+  async deploy(key: string, ensName: string) {
+    const ensArgs = this.ensService.argsFor(ensName);
+    if (ensArgs !== null) {
```
Comment (marekkirejczyk) [inline universal-login-relayer/lib/services/WalletService.ts]: `ensure(ensArgs !== null, InvalidENSDomain, ensName)`

File: universal-login-relayer/lib/services/WalletService.ts [hunk 5]
```diff
@@ -33,6 +36,17 @@ class WalletService {
...
+      const args = [key, ...ensArgs] as string[];
+      const initWithENS = encodeInitializeWithENSData(args);
+      const transaction = await this.factoryContract.createContract(key, initWithENS, {...defaultDeployOptions});
+      return transaction;
```
Comment (marekkirejczyk) [inline universal-login-relayer/lib/services/WalletService.ts]: `return await this.factoryContract.createContract(key, initWithENS, {...defaultDeployOptions});`

File: universal-login-relayer/test/relayer/routes/wallet.js [hunk 6]
```diff
@@ -47,6 +51,20 @@ describe('E2E: Relayer - WalletContract routes', async () => {
     expect(error).to.eq('Error: ENS domain marek.non-existing.eth does not exist or is not compatible with Universal Login');
   });
 
+  it('Counterfactual deployment', async () => {
```
Comment (marekkirejczyk) [inline universal-login-relayer/test/relayer/routes/wallet.js]: Perhaps add one edge case?
